packages/opencensus-core/src/stats/view.ts

```diff
@@ -32,32 +33,142 @@ export class BaseView implements View {
    * If no Tags are provided, then, all data is recorded in a single
    * aggregation.
    */
-  private rows: {[key: string]: AggregationData};
+  private rows: {[key: string]: AggregationData} = {};
+  /**
+   * A list of tag keys that represents the possible column labels
+   */
+  private columns: string[];
   /**
    * An Aggregation describes how data collected is aggregated.
    * There are four aggregation types: count, sum, lastValue and distirbution.
    */
   readonly aggregation: AggregationType;
   /** The start time for this view */
   readonly startTime: number;
+  /** The bucket boundaries in a Distribution Aggregation */
+  private bucketBoudaries?: number[];
   /**
    * The end time for this view - represents the last time a value was recorded
    */
   endTime: number;
   /** true if the view was registered */
-  registered: boolean;
+  registered = false;
 
+  /**
+   * Creates a new View instance. This constructor is used by Stats. User should
+   * prefer using Stats.createView() instead.
+   * @param name The view name
+   * @param measure The view measure
+   * @param aggregation The view aggregation type
+   * @param tagsKeys The Tags' keys that view will have
+   * @param description The view description
+   * @param bucketBoudaries The view bucket boundaries for a distribution
+   * aggregation type
+   */
   constructor(
       name: string, measure: Measure, aggregation: AggregationType,
-      tagKeys: string[], description?: string) {
-    throw new Error('Not Implemented');
+      tagsKeys: string[], description?: string, bucketBoudaries?: number[]) {
+    if (aggregation === AggregationType.DISTRIBUTION && !bucketBoudaries) {
+      throw new Error('No bucketBoundaries specified');
+    }
+    this.name = name;
+    this.description = description;
+    this.measure = measure;
+    this.columns = tagsKeys;
+    this.aggregation = aggregation;
+    this.startTime = Date.now();
+    this.bucketBoudaries = bucketBoudaries;
+  }
+
+  /**
+   * Records a measurement in the proper view's row. This method is used by
+   * Stats. User should prefer using Stats.record() instead.
+   *
+   * Measurements with measurement type INT64 will have its value truncated.
+   * @param measurement The measurement to record
+   */
+  recordMeasurement(measurement: Measurement) {
+    // Checks if measurements has all tags in views
+    for (const tagKey of this.columns) {
+      if (!Object.keys(measurement.tags).find((key) => key === tagKey)) {
```

Comment:
Prefer `some` over `find` here, as the return result could be an empty string. Of course any empty string key is not a real tag, but this will at least "work" if `some` is used instead.

Reply:
Great suggestion. Updated.